From b4571c292e76382f98266322a025c6383bd02f30 Mon Sep 17 00:00:00 2001
From: Phil <me@pnelson.ca>
Date: Fri, 15 Dec 2023 06:54:25 +0000
Subject: [PATCH 117/188] Add initial support for janet-lang (#9081)

* Add initial support for janet-lang

* Use default roots for janet-lang
---
 book/src/generated/lang-support.md   | 1 +
 languages.toml                       | 9 +++++++++
 runtime/queries/janet/highlights.scm | 1 +
 3 files changed, 11 insertions(+)
 create mode 100644 runtime/queries/janet/highlights.scm

--- a/book/src/generated/lang-support.md
+++ b/book/src/generated/lang-support.md
@@ -73,6 +73,7 @@
 | idris |  |  |  | `idris2-lsp` |
 | iex | ✓ |  |  |  |
 | ini | ✓ |  |  |  |
+| janet | ✓ |  |  |  |
 | java | ✓ | ✓ | ✓ | `jdtls` |
 | javascript | ✓ | ✓ | ✓ | `typescript-language-server` |
 | jinja | ✓ |  |  |  |
--- a/languages.toml
+++ b/languages.toml
@@ -2952,3 +2952,12 @@ file-types = ["log"]
 [[grammar]]
 name = "log"
 source = { git = "https://github.com/Tudyx/tree-sitter-log", rev = "62cfe307e942af3417171243b599cc7deac5eab9" }
+
+[[language]]
+name = "janet"
+scope = "source.janet"
+injection-regex = "janet"
+file-types = ["janet"]
+comment-token = "#"
+indent = { tab-width = 2, unit = "  " }
+grammar = "clojure"
--- /dev/null
+++ b/runtime/queries/janet/highlights.scm
@@ -0,0 +1 @@
+; inherits: clojure
